gh-92210: Port of Cinder's socket.__init__ which uses argument clinic. (#92237)
python/cpython: Misc/NEWS.d/next/Library/2022-05-03-17-33-46.gh-issue-92210.csDOQM.rst

> OPort of Cinder's ``socket.__init__`` which uses Argument Clinic to CPython.

Reviewer: ```suggestion
Port ``socket.__init__`` to Argument Clinic. Patch by Cinder.
```
